kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java

```diff
@@ -367,4 +369,16 @@ public static Duration getAge(HasMetadata kubernetesResource) {
     Instant instant = Instant.parse(kubernetesResource.getMetadata().getCreationTimestamp());
     return Duration.between(instant, Instant.now()).abs();
   }
+
+  public static <T extends HasMetadata> Class<? extends KubernetesResourceList> inferListType(Class<T> type) {
```

Comment:
Maybe that should be moved to `HasMetadata`?

Reply:
I'm not sure about this. There are several resources which `implement HasMetadata` but do not have a list type(For example `TokenReview`, `SelfSubjectAccessReview` etc)